penguin/llm/reasoning_variants.py

```
@@ -23,6 +24,32 @@ def _normalized_model_key(model_id: str) -> str:
     return re.sub(r"[^a-z0-9]+", "-", value).strip("-")
 
 
+def reasoning_effort_from_metadata(raw_value: Any) -> str | None:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `raw_value`

(ANN401)